vamp-server/convert.cpp

```diff
@@ -556,8 +558,15 @@ readInput(string format, RequestOrResponse::Direction direction, bool &eof)
     if (format == "json") {
         string err;
         auto result = readInputJson(direction, err, eof);
-        if (err != "") throw runtime_error(err);
-        else return result;
+        const bool isRecognisedError = !result.success && result.errorText != "";
+        
+        // if the RequestOrResponse (result) has been populated with success=false and an error message
+        // then the server returned a well formed error, it is safe to return it for conversion 
+        // -- but if err is populated, something else has gone wrong
+        if (isRecognisedError || err == "") 
```

Comment:
I realise my suggestion does the same thing, it was intended to add clarity (and was written seconds before you posted your comment with further clarification - not in response).

Well, I want to do the latter because the former is exactly what the code was before I changed it (though this time with braces). 

I wanted the short circuiting, because if there is a valid error, it needs to be written out. I honestly hadn't considered the case where `isRecognisedError` is false and `err` is non-empty. I really just needed a way to get the error written out and kept the checking of `err` in there in case that could happen otherwise - in which case throw an exception. Which is what this does.

I am still at a bit of a loss about what to actually do here.

Delete the comment? Aha.